pkg/services/hcloud/server/server.go

```diff
@@ -695,8 +708,8 @@ func (s *Service) findServer(ctx context.Context) (*hcloud.Server, error) {
 
 	servers, err := s.scope.HCloudClient.ListServers(ctx, opts)
 	if err != nil {
-		hcloudutil.HandleRateLimitExceeded(s.scope.HCloudMachine, err, "ListServers")
-		return nil, fmt.Errorf("failed to list servers: %w", err)
+		err := handleRateLimit(s.scope.HCloudMachine, err, "ListServers", "failed to list servers")
```

Comment:
you don't have to do it in two separate lines. You can just inline it and directly return the result of "handleRateLimit"